UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Webjob.UnitTests/Services/FulfilmentCallBackServiceTest.cs

```diff
@@ -0,0 +1,145 @@
+            fakeCallBackClient = A.Fake<ICallBackClient>();
+            fakeFileShareServiceConfig = Options.Create(new FileShareServiceConfiguration(){ });
+            fakeLogger = A.Fake<ILogger<FulfilmentCallBackService>>();
+
+            fulfilmentCallBackService = new FulfilmentCallBackService(fakeEssCallBackConfiguration, fakeCallBackClient, fakeFileShareServiceConfig, fakeLogger);
+        }
+
+        #region GetCallBackResponse
+        public CallBackResponse GetCallBackResponse()
+        {
+            return new CallBackResponse() 
+            {
+                Specversion="1.0",
+                Type = "test",
+                Source = "test",
+                Subject = "test",
+                Time = DateTime.UtcNow.ToString("yyyy-MM-ddTHH:mm:ssZ", CultureInfo.InvariantCulture),
+                DataContentType = "application/json",
+                Data = new ExchangeSetResponse()
+            };
+        }
+        #endregion
+
+        #region GetSalesCatalogueServiceResponse
+        private static SalesCatalogueProductResponse GetSalesCatalogueServiceResponse()
+        {
+            return new SalesCatalogueProductResponse()
+            {
+                ProductCounts = new ProductCounts()
+                {
+                    RequestedProductCount = 12,
+                    RequestedProductsAlreadyUpToDateCount = 5,
+                    RequestedProductsNotReturned = new List<RequestedProductsNotReturned>
+                    {
+                        new RequestedProductsNotReturned()
+                        {
+                            ProductName = "test",
+                            Reason = "notfound"
+                        }
+                    },
+                    ReturnedProductCount = 4
+                },
+                Products = new List<Products> {
+                            new Products {
+                                ProductName = "DE5NOBRK",
+                                EditionNumber = 0,
+                                UpdateNumbers = new List<int?> {0,1},
+                                FileSize = 400
+                            }
+                        }
+            };
+        }
+        #endregion
+
+        #region GetScsResponseQueueMessage
+        private SalesCatalogueServiceResponseQueueMessage GetScsResponseQueueMessage()
+        {
+            return new SalesCatalogueServiceResponseQueueMessage
+            {
+                BatchId = "7b4cdf10-adfa-4ed6-b2fe-d1543d8b7272",
+                FileSize = 4000,
+                ScsResponseUri = "https://test/ess-test/7b4cdf10-adfa-4ed6-b2fe-d1543d8b7272.json",
+                CallbackUri = "https://test-callbackuri.com",
+                CorrelationId = "727c5230-2c25-4244-9580-13d90004584a"
+            };
+        }
+        #endregion
+
+        [Test]
+        public async Task WhenValidCallBackUriInRequest_ThenSendCallBackResponseReturnsTrue()
+        {
```

Comment:
This test should really be called WhenCallBackUriInRequest_ThenSendCallBackResponseReturnsTrue as it only really tests that the Uri is not null or white space.

Reply:
as per suggestion, i have done required changes.